climada/util/plot.py

```diff
@@ -290,6 +292,98 @@ def geo_im_from_array(array_sub, coord, var_name, title,
 
     return axes
 
+def geo_scatter_categorical(array_sub, geo_coord, var_name, title,
```

Comment:
This new method is copied from `geo_scatter_from_array`. The main differences are:

- Addition or argument `cat_name`
- Docstring
- Cmap definition in lines 335 - 339
- Definition of discrete (not continuous) cmap in lines 361 - 366
- Plot in line 369 adds `norm` and `cmap` arguements
- Add ticks label to colorbar in 377 -379

Reply:
If it's a variation of that other function, then maybe it's easier to maintain the code if you don't create a new function but just add a new parameter to the existing function that allows to change its behavior.

Reply:
That could be an idea indeed... at the same time, too many different behaviours for the same function are also hard to maintain right? Not sure what is best here.

Reply:
I also think it would make sense to not copy the code from the other function. Isn't it also possible to call the original function in the new function after creating the colormap and the category data? Or would it slow it to much?

Reply:
Also a good idea. So there are three possible ways I see:

1. Keep as is, two separate methods (current implementation)
2. Add  a categorical  boolean argument to `geo_scatter_from_array` (suggested by @tovogt )
3. Make the `geo_scatter_categorical` method only build the colormap and then call `geo_scatter_array` (suggested by @zeliest)

An important design question that remains independently of the choices is:
- should the categories be shared among all subplots, or should each subplots have its own categories?
Example:
array1 = [1, 2, 1, 1]
array2 = [1, 2, 3, 1,  1]
array3 = ['a', 'b']
Shared categories: [1, 2, 3, 'a', 'b']
Categories per subplots: [1, 2] and [1, 2, 3] and ['a', 'b']

Reply:
I think it might become too much information with different categories per subplots. And you could always make a separate plot right if you wanted different categories?

Reply:
Yes, this is correct. So you recommend the shared categories approach? Also, which one of the three implementations do you favor?

Reply:
yes shared categories, and I guess I would personnaly do it how I suggested (3), but I cannot say it is the best way :)